pytorch_lightning/trainer/data_loading.py

```diff
@@ -133,20 +137,23 @@ def auto_add_sampler(self, dataloader: DataLoader, shuffle: bool) -> DataLoader:
                 )
 
             # replace with distributed sampler
-            sampler = self._get_distributed_sampler(dataloader, shuffle)
-            dataloader = self.replace_sampler(dataloader, sampler)
+            sampler = self._get_distributed_sampler(dataloader, shuffle, mode=mode)
+            dataloader = self.replace_sampler(dataloader, sampler, mode=mode)
 
         return dataloader
 
     @staticmethod
-    def _resolve_batch_sampler(dl_args, dataloader, sampler):
+    def _resolve_batch_sampler(dl_args, dataloader, sampler, mode: Optional[RunningStage] = None) -> Dict[str, Any]:
         batch_sampler = getattr(dataloader, "batch_sampler")
-        if batch_sampler is not None and type(batch_sampler) is not BatchSampler:
+        is_predicting = mode == RunningStage.PREDICTING
+        if (batch_sampler is not None and type(batch_sampler) is not BatchSampler) or is_predicting:
             batch_sampler = type(batch_sampler)(
                 sampler,
                 batch_size=batch_sampler.batch_size,
-                drop_last=batch_sampler.drop_last,
+                drop_last=False if is_predicting else batch_sampler.drop_last,
```

Comment:
```suggestion
                drop_last=(False if is_predicting else batch_sampler.drop_last),
```
so the user returning a dataloader with `drop_last=True for` for prediction will be ignored?

Reply:
Yes, by default the trainer will make a prediction for all available samples in the dataset.